src/cli/init.js

```diff
@@ -0,0 +1,158 @@
+// @flow
+// Implementation of `sourcecred init`
+
+import dedent from "../util/dedent";
+import {type RepoId, stringToRepoId} from "../core/repoId";
+import {type Project, projectToJSON} from "../core/project";
+import type {Command} from "./command";
+import * as Common from "./common";
+import fs from "fs-extra";
+import process from "process";
+import read from "read";
+import path from "path";
+import {fetchGithubOrg} from "../plugins/github/fetchGithubOrg";
+import {type DiscourseServer} from "../plugins/discourse/loadDiscourse";
+
+function usage(print: (string) => void): void {
+  print(
+    dedent`\
+    usage: sourcecred init
+           sourcecred init --help
+
+    Sets up a new SourceCred instance, by interactively creating a SourceCred
+    project configuration, and saving it to 'sourcecred.json'.
+
+    Arguments:
+        --help
+            Show this help message and exit, as 'sourcecred help init'.
+
+    Environment variables:
+        SOURCECRED_GITHUB_TOKEN
+            API token for GitHub. This should be a 40-character hex
+            string. Required if you want to load whole GitHub orgs.
+
+            To generate a token, create a "Personal access token" at
+            <https://github.com/settings/tokens>. When loading data for
+            public repositories, no special permissions are required.
+            For private repositories, the 'repo' scope is required.
+    `.trimRight()
+  );
+}
+
+function die(std, message) {
+  std.err("fatal: " + message);
+  std.err("fatal: run 'sourcecred help init' for help");
+  return 1;
+}
+
+const initCommand: Command = async (args, std) => {
+  for (let i = 0; i < args.length; i++) {
+    switch (args[i]) {
+      case "--help": {
+        usage(std.out);
+        return 0;
+      }
+      default: {
+        return die(std, `Unexpected argument ${args[i]}`);
+      }
+    }
+  }
+  const dir = process.cwd();
+  const projectFilePath = path.join(dir, "sourcecred.json");
+  if (await fs.exists(projectFilePath)) {
+    return die(std, `Refusing to overwrite sourcecred.json file in ${dir}`);
+  }
+  const basename = path.basename(dir);
+  const name = await aread({
+    prompt: "Choose an instance name:",
+    default: basename,
+  });
+  std.out("");
+
+  let repoIds: RepoId[] = [];
+  const githubToken = Common.githubToken();
+  if (githubToken == null) {
+    std.out("Skipping organization loading, as no GitHub token available.");
```

Comment:
We might also fall back on using interaction for the token, as a new user might overlook the token requirement. It would also allow us to mix org and repo adding in a single loop, so people don't have to plan out the order ahead of time.